src/gobupload/storage/relate.py

```diff
@@ -345,6 +347,115 @@ def _get_select_from(dst_fields, dst_match_fields, src_fields, src_match_fields)
     return select_from
 
 
+def _query_missing(query, items_name, max_warnings=50):
+    """
+    Query for anu missing attributes
+
+    :param query: query to execute
+    :param items_name: name of the missing attribute
+    :return: None
+    """
+    count = {
+        'current': 0,
+        'historic': 0
+    }
+    for data in _get_data(query):
+        period = 'current' if data.get('eind_geldigheid') is None else 'historic'
+        if count[period] < max_warnings:
+            msg = f"{period} {items_name}"
+            logger.warning(msg, {
+                'id': msg,
+                'data': {k: v for k, v in data.items() if v is not None}
+            })
+        count[period] += 1
+        if count[period] == max_warnings:
+            logger.warning(f"Too many (>{max_warnings}) {items_name}")
+        if count['current'] >= max_warnings and count['historic'] >= max_warnings:
+            break
+
+
+def check_relations(src_catalog_name, src_collection_name, src_field_name):
+    """
+    Check relations for any dangling relations
+
+    Dangling can be because a relation exist without any bronwaarde
+    or the bronmwaarde cannot be matched with any referenced entity
+
+    :param src_catalog_name:
+    :param src_collection_name:
+    :param src_field_name:
+    :return: None
+    """
+    # Get the source catalog, collection and field for the given names
+    model = GOBModel()
+
+    src_table_name = model.get_table_name(src_catalog_name, src_collection_name)
+    rel_name = get_relation_name(model, src_catalog_name, src_collection_name, src_field_name)
+    rel_table_name = f"rel_{rel_name}"
+
+    src_has_states = model.has_states(src_catalog_name, src_collection_name)
+
+    select = ["_id"]
+    where = [f"src_id = {src_table_name}._id"]
+    if src_has_states:
+        select.extend(["volgnummer", "begin_geldigheid", "eind_geldigheid"])
+        where.extend([f"src_volgnummer = {src_table_name}.volgnummer"])
+    select = ",\n    ".join(select)
+    where = " AND\n            ".join(where)
+
+    # select all relations that do not have an entry in the relations table
+    srcs_without_relations = f"""
+SELECT
+    {select}
+FROM
+    {src_table_name}
+WHERE
+    _date_deleted IS NULL AND
+    {src_field_name} ->> 'bronwaarde' IS NOT NULL AND
```

Comment:
Isn't this always true? Within a JSON object the NULL value is a string 'null', which is also NOT NULL.

Reply:
Add comment in code to explain behaviour